torch/_export/serde/serialize.py

```diff
@@ -1243,7 +1243,7 @@ def serialize_treespec(self, treespec):
         def store_namedtuple_fields(ts):
             if ts.type is None:
                 return
-            if ts.type == namedtuple:
+            if ts.type is namedtuple or pytree.is_namedtuple_class(ts.type):
```

Comment:
NB: Due to this, previously, all tests passed in OSS while break internally. With the latest commit, this can be reverted.